### ISSUE TITLE
SC2034 - consier a "test -v" as usage/reference

### ISSUE THREAD

**Issue by pikrzysztof** [koalaman/shellcheck#2927]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or "online"):  online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash

#!/bin/bash
x=something
if test -v x; then
        echo set
else
        echo unset
fi     
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
[Line 3:](javascript:setPosition(3, 1))
x=something
^-- [SC2034](https://www.shellcheck.net/wiki/SC2034) (warning): x appears unused. Verify use (or export if used externally).
```

#### Here's what I wanted or expected to see:

no output. `x` is used/referenced. 

My use-case is following:

```bash
while [[ "$#" -gt 0 ]] ; do
    case "$1" in
	-n) readonly nflag=true; shift 1;;
        *) usage;;
    esac
done
if test -v nflag; then
    do_n_stuff
else
    do_non_n_stuff
fi
```
